### PR TITLE
[distroless] move cloud-controller-manager binaries to distroless

### PR DESCRIPTION
**Pull request by universal-itengineer**
## Description

cloud-controller-manager binaries use distroless images and built from our repos instead of using alpine:go images.
- [x] cloud-provider-aws
- [x] cloud-provider-azure
- [x] cloud-provider-gcp
- [x] cloud-provider-yandex
- [x] ee/cloud-provider-openstack
- [x] ee/cloud-provider-vsphere

## Why do we need it, and what problem does it solve?

This PR is a part of moving Deckhouse to distroless.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: cloud-provider-aws
type: chore
summary: "`cloud-controller-manager` use distroless based image."
impact_level: default
---
section: cloud-provider-azure
type: chore
summary: "`cloud-controller-manager` use distroless based image."
impact_level: default
---
section: cloud-provider-gcp
type: chore
summary: "`cloud-controller-manager` use distroless based image."
impact_level: default
---
section: cloud-provider-yandex
type: chore
summary: "`cloud-controller-manager` use distroless based image."
impact_level: default
---
section: ee/cloud-provider-openstack
type: chore
summary: "`cloud-controller-manager` use distroless based image."
impact_level: default
---
section: ee/cloud-provider-vsphere
type: chore
summary: "`cloud-controller-manager` use distroless based image."
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by z9r5** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 6351209586 6351209586-1-con-1-25 /sys/deckhouse-oss/install:pr5985 centos@3.74.49.178

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by universal-itengineer** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 6376972203 6376972203-1-con-1-25 /sys/deckhouse-oss/install:pr5985 centos@35.156.149.44

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
Please add https://github.com/deckhouse/deckhouse/tree/main/ee/modules providers

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/openstack'...

**Comment by z9r5** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 6376972203 6376972203-1-con-1-25 /sys/deckhouse-oss/install:pr5985 centos@35.156.149.44

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by universal-itengineer** [comment]
/e2e/abort vsphere;Standard;containerd;1.25 6394012131 6394012131-1-con-1-25 /sys/deckhouse-oss/install:pr5985 ubuntu@10.220.203.29

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by z9r5** [comment]
/e2e/abort vsphere;Standard;containerd;1.25 6402998204 6402998204-1-con-1-25 /sys/deckhouse-oss/install:pr5985 ubuntu@10.220.203.27

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/vsphere'...

File: ee/modules/030-cloud-provider-openstack/images/cloud-controller-manager/werf.inc.yaml [hunk 1]
```diff
@@ -53,5 +45,12 @@ shell:
   - cd /src
   - test -d /patches/{{ $version }} && for patchfile in /patches/{{ $version }}/*.patch ; do patch -p1 < ${patchfile}; done
   - make VERSION={{ $value.ccm.openstack }} GIT_VERSION={{ $value.ccm.openstack }} openstack-cloud-controller-manager
+  {{- if semverCompare ">=1.27" $version }}
+  - chown 64535:64535 /src/openstack-cloud-controller-manager
+  - chmod 0755 /src/openstack-cloud-controller-manager
+  {{- else }}
+  - chown 64535:64535 /src/openstack-cloud-controller-manager-amd64
+  - chmod 0755 /src/openstack-cloud-controller-manager-amd64
+  {{- end }}
```
Comment (RomanenkoDenys) [inline ee/modules/030-cloud-provider-openstack/images/cloud-controller-manager/werf.inc.yaml:54]: strange construction, simply move openstack-cloud-controller-manager-amd64 to openstack-cloud-controller-manager

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
one simple question

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @universal-itengineer. I've started the workflow for label 'e2e/run/openstack'...

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Pull request merged**